content/topics/Analyze/causal-inference/panel-data/event_studies.md

```diff
@@ -0,0 +1,263 @@
+
+As you can see, at the tails of event time we are dealing with very few observations (as low as 5). 
+
+<p align = "center">
+<img src = "../images/event_hist.png" width="500">
+</p>
+
+This could impose a problem as event time dummies with few observations cannot be estimated. 
+To deal with the low number of observations in the tails, we can resort to the technique of **binning**. 
+The solution is creating bins with multiple event times at the tails. In this case we create two bins containing the extreme values of 'eventtime':
+
+{{% codeblock %}}
+```R
+# Update the 'waste' dataframe by adding a new column 'eventtime_bin'
+waste <- waste %>%
+  
+  # Create a new column 'eventtime_bin':
+  # If 'eventtime' is less than or equal to -37, set 'eventtime_bin' to -37.
+  # If 'eventtime' is greater than or equal to 28, set 'eventtime_bin' to 28.
+  # Otherwise, set 'eventtime_bin' to 'eventtime' itself.
+  mutate(eventtime_bin = ifelse(eventtime <= -37, -37,
+                                ifelse(eventtime >= 28, 28, eventtime)))
+
+```
+{{% /codeblock %}}
+
+
+To estimate the event-time model, we use the function feols from fixest. The full syntax is: modelname = feols(outcome ~ i(factor_var, f, ref, drop, keep) | groupFE + weekFE, data=namedataset). Thanks to the i() we can easily include interaction terms. 
+
+{{% tip %}}
+When running fixed-effects models, it is handier to use the command feols, part of the package fixest. Feols allows for very easy estimation of event time models.
+{{% /tip %}}
+
+The term i() interacts the variable factor_var, which is going to be the event-time variable eventtime_bin, with a dummy variable for each value in f. The term f stands for all groups that are (eventually) treated. That holds for all groups (i.e. routes) in our data, thus we simply define a variable that contains only ones:
+
+{{% codeblock %}}
+```R
+waste$allones <- 1
+```
+{{% /codeblock %}}
+
+Let's now estimate the model. 
+
+{{% codeblock %}}
+```R
+mod_twfe = feols(residual_weight ~ i(eventtime_bin, allones, ref=0) | route + calendar_week, data=waste)
+etable(mod_twfe)
+```
+{{% /codeblock %}}
+
+The argument ref is the value of event-time taken as a reference. We take it to be 0, i.e. the last ‘normal’ period, the last period before the start of the treatment.
+
+You can now see a long list of coefficients $\alpha_\tau$ which can be pretty confusing to go through, let's plot them with respect to time to have a straightforward picture of treatment effect. 
+
+{{% codeblock %}}
+```R
+iplot(mod_twfe,
+      xlab = 'Time to treatment',
+      main = 'Event study: Staggered treatment (TWFE)', 
+      type = "l", 
+      col = "blue", 
+      grid = TRUE,
+      pch = 1, 
+      legend = TRUE,
+      legend.text = c("Treatment Effect", "95% CI"),
+      ylab = "Treatment Effect",
+      lty = 2, 
+      lwd = 1)
+```
+{{% /codeblock %}}
+
+<p align = "center">
+<img src = "../images/event_timeplot.png" width="500">
+</p>
+
+
```

Comment:
You may want to add a disclaimer that here you assume treatment effects to be homogenous across time and groups. The estimated event time coeffs are based on comparisons of newly treated groups (routes) relative to routes that have not yet been treated as well as comparisons of newly treated groups relative to _already treated groups._ If at all treatment effects are heterogenous, comparing newly treated with already treated (so called "forbidden comparisons") can lead to bias in the treatment effect. One would then use the Bacon decomposition method proposed by Goodman-Bacon, 2021 to perform a diagnosis as to whether this is the case. This is an important step before going into a robust methods to overcome the bias. Given that we dont have a BB on bacon decomposition yet I think it would be nice if you could add it in this one. If you find that there is no bias then you can conclude that your specification is robust to this.

Reply:
Here's the link to the paper by Andrew Bacon: Goodman-Bacon, A. (2021). Difference-in-differences with variation in treatment timing. Journal of econometrics, 225(2), 254-277.

Reply:
`bacondecomp` package can be used in R

Reply:
@lachlandeer @shrabasteebanerjee Let me know what you think about adding bacon decomposition part to this BB?

Reply:
i'd be cautious about "adding too much" to one topic. It gets overwhelming or gets explained in too much of a hurry to be clear to a reader.